Microsoft.Toolkit.Uwp.UI/Extensions/Common/StringExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Toolkit.Uwp.UI.Extensions.Common
+{
+    /// <summary>
+    /// All common string extensions should go here
+    /// </summary>
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Returns whether said string is a valid email or not.
+        /// Uses general Email Regex (RFC 5322 Official Standard) from emailregex.com
+        /// </summary>
+        /// <returns><c>true</c> for valid email.<c>false</c> otherwise</returns>
+        public static bool IsEmail(this string str)
```

Comment:
The TextBoxRegEx implementation should also be updated to use this new method now too.

https://github.com/Microsoft/UWPCommunityToolkit/blob/master/Microsoft.Toolkit.Uwp.UI/Extensions/TextBoxRegEx/TextBoxRegex.cs#L91

Reply:
I can do another PR for moving everything related to `TextBoxRegEx` to this helper method. For now I will update this PR moving the string constants separate.